server/paymentProviders/opencollective/collective.js

```diff
@@ -76,7 +76,7 @@ paymentMethodProvider.processOrder = async order => {
     CollectiveId: order.CollectiveId,
     PaymentMethodId: order.PaymentMethodId,
     type: TransactionTypes.CREDIT,
-    kind: TransactionKind.CONTRIBUTION,
+    kind: order.data?.customData?.isBalanceTransfer ? TransactionKind.BALANCE_TRANSFER : TransactionKind.CONTRIBUTION,
```

Comment:
Maybe it's time to introduce an `Order.kind`. The possible values would be `ADDED_FUNDS`, `CONTRIBUTION` and `BALANCE_TRANSFER`.

What do you think @Betree ?

Reply:
That would be cleaner than a flag in `data`. I'm wondering though, do we really need an order for added funds and balance transfers? What would be the downsides of creating the transactions directly, without an order?

Reply:
I don't know, we're using Orders because this is the way it is until now.

Because of this, we can use the GraphQL `createOrder` mutation. If we're reconsidering this, we would have to create a separate mutation for `BALANCE_TRANSFER` (we already have one for ADDED_FUNDS).

I actually appreciate having an Order or an Expense for most Transactions. But looking at it more closely, I don't see a real value today.

Reply:
Also, having an order seems required to use the complex PaymentMethod code which create the transactions. Maybe it's best to keep Order for now, out of simplicity.

Reply:
So I think for now we will keep this as it is. But note that I've removed the `isBalanceTransfer` field from `customData` as suggested at https://github.com/opencollective/opencollective-frontend/pull/6761#discussion_r691861458